### PR TITLE
Add realm specific implementation for List.add and List.insert

### PR DESCRIPTION
**Pull request by nielsenko**

### CODE REVIEW

**Review by nirinchev — APPROVED** [review]

File: test/list_test.dart [hunk 1]
```diff
@@ -699,4 +699,15 @@ Future<void> main([List<String>? args]) async {
     expect(team.players, isNot(players));
     expect(team.players, unorderedMatches(players));
   });
+
+  test('ManagedRealmList.length= throws', () {
```
Comment (blagoev) [inline test/list_test.dart:703]: should we have a test for length decrease as well. Now the `removeRange` path is not tested

**Review by blagoev — APPROVED** [review]
I think we should add a test for the `removeRange` path of length setter

**Comment by nielsenko** [comment]
> I think we should add a test for the `removeRange` path of length setter

I have added the requested test

File: lib/src/list.dart [hunk 2]
```diff
@@ -63,7 +63,11 @@ class ManagedRealmList<T extends Object?> extends collection.ListBase<T> with Re
   @override
```
Comment (desistefanova) [inline lib/src/list.dart:63]:
```suggestion
  /// Setting the `length` is a required method on [List], but makes less sense
  /// for [RealmList]s. You can only decrease the length, increasing it throws an [RealmException].
```

**Pull request merged**